File: be/internal/models/payments.go
package models
import (
	"encoding/json"
	"github.com/google/uuid"
	"time"
)
type PaymentOrder struct {
	OrderID            uuid.UUID       `json:"order_id"`
	PublicToken        string          `json:"-"`
	UserID             uuid.UUID       `json:"user_id"`
	UserEmail          string          `json:"user_email"`
	UserDisplayName    string          `json:"user_display_name"`
	Provider           string          `json:"provider"`
	PlanCode           string          `json:"plan_code"`
	BaseAmount         int64           `json:"base_amount"`
	Amount             int64           `json:"amount"`
	TipAmount          int64           `json:"tip_amount"`
	Currency           string          `json:"currency"`
	SubscriptionType   string          `json:"subscription_type"`
	Status             string          `json:"status"`
	ProviderStatus     string          `json:"provider_status"`
	ProviderPaymentID  string          `json:"provider_payment_id"`
	ProviderErrorCode  string          `json:"provider_error_code"`
	ProviderMessage    string          `json:"provider_message"`
	ProviderResponse   json.RawMessage `json:"provider_response"`
	ProviderFeePercent float64         `json:"provider_fee_percent"`
	FeeAmount          int64           `json:"fee_amount"`
	NetAmount          int64           `json:"net_amount"`
	AccessFrom         *time.Time      `json:"access_from,omitempty"`
	AccessUntil        *time.Time      `json:"access_until,omitempty"`
	ReturnTo           string          `json:"return_to"`
	SuccessURL         string          `json:"success_url"`
	FailURL            string          `json:"fail_url"`
	PaymentURL         string          `json:"payment_url"`
	LastCheckedAt      *time.Time      `json:"last_checked_at,omitempty"`
	NotifiedAt         *time.Time      `json:"notified_at,omitempty"`
	PaidAt             *time.Time      `json:"paid_at,omitempty"`
	FailedAt           *time.Time      `json:"failed_at,omitempty"`
	RefundedAt         *time.Time      `json:"refunded_at,omitempty"`
	CreatedAt          time.Time       `json:"created_at"`
	UpdatedAt          time.Time       `json:"updated_at"`
}

File: fe/shared/types/payments.ts
export type PaymentOrderView = {
  order_id: string
  user_id: string
  user_email: string
  user_display_name: string
  plan_code: 'pro' | 'donation' | string
  base_amount: number
  amount: number
  tip_amount: number
  currency: string
  subscription_type: 'one_time' | 'recurring' | string
  status: 'pending' | 'success' | 'failed' | 'refunded' | 'canceled' | string
  provider: string
  provider_status: string
  provider_payment_id: string
  provider_error_code: string
  provider_message: string
  provider_fee_percent: number
  fee_amount: number
  net_amount: number
  access_from?: string | null
  access_until?: string | null
  return_to: string
  payment_url: string
  last_checked_at?: string | null
  notified_at?: string | null
  paid_at?: string | null
  failed_at?: string | null
  refunded_at?: string | null
  created_at: string
  updated_at: string
}
export type PaymentAccessSummary = {
  has_active_access: boolean
  plan_code: string
  amount: number
  tip_amount: number
  currency: string
  access_from?: string | null
  access_until?: string | null
  order_id?: string | null
  latest_order?: PaymentOrderView | null
}
export type PaymentCreateOrderResult = {
  order: PaymentOrderView
  payment_url: string
}
export type PaymentPublicLookupResult = {
  order: PaymentOrderView
}
export type PaymentAdminOrdersSummary = {
  orders_total: number
  orders_success: number
  orders_pending: number
  orders_failed: number
  orders_refunded: number
  paid_users_total: number
  patron_users_total: number
  active_access_users: number
  gross_revenue: number
  net_revenue: number
  tip_revenue: number
  mrr: number
  churn_rate: number
  patron_share: number
  last_successful_paid?: string | null
}
